opa-audit-operator/docs/gatekeeper-rb.yaml.template

```diff
@@ -0,0 +1,44 @@
+
+# If RBAC is enabled in the cluster, then the operator pod for both
+# metallb-controller and metallb-speaker need to be granted permission to # use the K8s API for some specific actions.
+# This step is not automated because once bug LP:1896076 and LP:1886694 are fixed, it will no
+# longer be necessary to use the K8s API.
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: use-k8s-api
+rules:
+- apiGroups: ["policy", "rbac.authorization.k8s.io", "apiextensions.k8s.io"]
+  resources: ["podsecuritypolicies", "roles", "rolebindings", "customresourcedefinitions"]
+  verbs: ["create", "delete"]
+- apiGroups: ["config.gatekeeper.sh"]
+  resources: ["configs"]
+  verbs: ["create", "delete"]
+- apiGroups: [""]
+  resources: ["configmaps"]
+  verbs: ["get", "list", "watch"]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: use-k8s-api
+subjects:
+- kind: ServiceAccount
+  name: gatekeeper-manager-operator
+  # change namespace name according to your environment
+  namespace: gatekeeper
+- kind: ServiceAccount
+  namespace: gatekeeper
+  name: gatekeeper-audit-operator
+  # change namespace name according to your environment
+  namespace: metallb-system
+- kind: User
+  name: {{service_account_user}}
+  # change namespace name according to your environment
+  namespace: metallb-system
+roleRef:
+  kind: ClusterRole
+  name: use-k8s-api
+  apiGroup: rbac.authorization.k8s.io
+---
```

Comment:
The `---` at the end of this file is causing a test failure:

```
____________________________________________ test_build_and_deploy _____________________________________________
Traceback (most recent call last):
  File "/home/gkk/work/tasks/review/opa-gatekeeper-operators/opa-audit-operator/tests/integration/test_charm.py", line 39, in test_build_and_deploy
    if k8s_obj["kind"] == "ClusterRoleBinding":
TypeError: 'NoneType' object is not subscriptable
```

Can you remove it?

Reply:
updated